analyzer/codechecker_analyzer/analyzers/cppcheck/analyzer.py

```diff
@@ -0,0 +1,350 @@
+
+            # Cppcheck runs with all checkers enabled for the time being
+            # the unneded checkers are passed as suppressed checkers
+            analyzer_cmd.append('--enable=all')
+
+            if enabled_severity_levels:
+                analyzer_cmd.append('--enable=' +
+                                    ','.join(enabled_severity_levels))
+
+            for checker_name, value in config.checks().items():
+                if value[0] == CheckerState.disabled:
+                    if checker_name.startswith("cppcheck-"):
+                        checker_name = checker_name[9:]
+                    # TODO python3.9 removeprefix method is better than lstrip
+                    analyzer_cmd.append('--suppress=' + checker_name)
+
+            # unusedFunction check is for whole program analysis,
+            # which is not compatible with per source file analysis.
+            analyzer_cmd.append('--suppress=unusedFunction')
+
+            # Add extra arguments.
+            analyzer_cmd.extend(config.analyzer_extra_arguments)
+
+            # Add includes.
+            for analyzer_option in self.buildaction.analyzer_options:
+                if analyzer_option.startswith("-I") or \
+                        analyzer_option.startswith("-U") or \
+                        analyzer_option.startswith("-D"):
+                    analyzer_cmd.extend([analyzer_option])
+                elif analyzer_option.startswith("-std"):
+                    standard = analyzer_option.split("=")[-1] \
+                        .lower().replace("gnu", "c")
+                    analyzer_cmd.extend(["--std=" + standard])
+
+            # By default there is no platform configuration,
+            # but a platform definition xml can be specified.
+            if 'platform' in config.analyzer_config:
+                analyzer_cmd.append(
+                        "--platform=" + config.analyzer_config['platform'][0])
+            else:
+                analyzer_cmd.append("--platform=native")
+
+            if 'addons' in config.analyzer_config:
+                for addon in config.analyzer_config["addons"]:
+                    analyzer_cmd.extend(
+                            ["--addon=" + str(Path(addon).absolute())])
+
+            if 'libraries' in config.analyzer_config:
+                for lib in config.analyzer_config["libraries"]:
+                    analyzer_cmd.extend(
+                            ["--library=" + str(Path(lib).absolute())])
+
+            # Cppcheck does not handle compiler includes well
+            # for include in self.buildaction.compiler_includes:
+            #    analyzer_cmd.extend(['-I',  include])
+
+            # TODO Suggest a better place for this
+            # cppcheck wont create the output folders for itself
+            output_dir = Path(result_handler.workspace, "cppcheck",
+                              result_handler.buildaction_hash)
+            output_dir.mkdir(exist_ok=True)
```

Comment:
Use `os.makedirs` instead, so it creates the parent directories, and then the other call might truly be redundant.